doc/user/sql/tail.md

```diff
@@ -36,6 +37,15 @@ Field | Represents
 `diff value` | Whether the record is an insert (`1`), delete (`-1`), or update (delete for old value, followed by insert of new value).
 `logical timestamp` | Materialize's internal logical timestamp.
 
+### AS OF
+
+If a timestamp is explicitly requested via `AS OF`, that timestamp will be used to determine what events to report rather than one automatically determined by Materialize.
+The timestamp provided to `AS OF` can be a number or a `TIMESTAMP` or `TIMESTAMPTZ` value.
+
+### WITH SNAPSHOT
+
+If `WITH SNAPSHOT` is specified, a snapshot of the state of the system at the chosen timestamp will be reported in addition to any subsequent events.
```

Comment:
This causes me to realize that we should maybe make a mention somewhere about Materialize choosing timestamps in the first place. That might make the `AS OF` section easier to write.

For this section, does this mean that you get a snapshot of the values with the greatest logical timestamp, or that you get all updates that occurred up to the greatest logical timestamp?